### PR TITLE
Release v0.7.0

### PR DESCRIPTION
**Pull request by thanodnl**
This PR releases v0.7.0 for ringpop-go.

Highlights:
* Labels support
* Fixed race conditions
* Better local state management

Please review `CHANGES.md` for the changelog.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,19 @@
 ringpop-go changes
 ==================
 
+v0.7.0
+------
+
+* Feature: Added label support to ringpop for nodes to annotate them self with more information #167 #171 #172
```
Comment (mennopruijssers) [inline CHANGES.md]: s/them self/themselves/

File: CHANGES.md [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+* Refactored internal representation of member state for more flexible reincarnation and state change of a member #159 #161
+* Make sure ringpop is listening for requests before bootstrapping #176 #146
+* Added support for imports in `.thrift` files in generated code for thrift forwarding #162
+* Mark generated code as being generated for suppression in diffs #169
+* Be more specific in the functionality required from TChannel #166
+* Tests: Run all tests on go 1.7 #179
+* Tests: More stable unit tests, integration tests and race detector tests #164 #178 #181 #182
```
Comment (mennopruijssers) [inline CHANGES.md]: This isn't only changing tests right? Maybe "promote" the races part to a normal line?

**Comment by Motiejus** [comment]
Maybe we can merge the two travis-ci runs? That would shrink the matrix even more!

* `RUN="make test-unit test-examples lint coveralls"`
* `RUN="make test-race"`

**Comment by thanodnl** [comment]
The end goal is to add `test-race` to the other entry, but for now keep it separate to validate it never errors again.

File: CHANGES.md [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+### Release notes
+
+#### Change to ringpop interface
+
+The ringpop interface changed two existing functions `GetReachableMembers` and
+`CountReachableMembers` that now take a variadic argument of type
+`swim.MemberPredicate` instead of no arguments. This does not change the usage
+of these functions, but does change the type of the function. This might cause
+custom declared interfaces to not match ringpop anymore. The solution is to
+change these functions in the interface used to match the current signature.
```
Comment (Motiejus) [inline CHANGES.md]: It would be nice to give an example of "before" and "after" code changes required?

File: CHANGES.md [hunk 4]
```diff
@@ -1,6 +1,37 @@
...
+
+#### Deprecated RegisterListener
+
+Due to a refactor in how event emitting is done the `RegisterListener` method is
+deprecated and starts logging warnings when used, instead the use of
+`AddListener` is advised. This function also returns if the listener has been
+added or not.
```
Comment (Motiejus) [inline CHANGES.md]: Also, specify whether the old function will keep working + emit warnings, or only emit warnings.

**Comment by mennopruijssers** [comment]
> I'm Menno, and I approve this PR
Reference to [SBYA](https://en.wikipedia.org/wiki/Stand_by_Your_Ad_provision) ;)

File: CHANGES.md (hunk 3 above)
Comment (thanodnl) [inline CHANGES.md]: I have added the actual old and new signatures below.

**Pull request merged**